dotcom-rendering/src/web/browser/hydration/doHydration.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import { hydrate, h } from 'preact';
+import { initPerf } from '../initPerf';
+
+/**
+ * This function dynamically imports and then hydrates a specific component in
+ * a specific part of the page
+ *
+ * @param name The name of the component we want to hydrate
+ * @param data The deserialised props we want to use for hydration
+ * @param marker The location on the DOM where the component to hydrate exists
+ */
+export const doHydration = (name: string, data: any, marker: HTMLElement) => {
+	const { start, end } = initPerf(`hydrate-${name}`);
+	start();
+	window
+		.guardianPolyfilledImport(
+			/* webpackInclude: /\.importable\.(tsx|jsx)$/ */
+			`../../components/${name}.importable`,
```

Comment:
I'd still like to find an improved way to do this step that doesn't depend on a static path to import components

Reply:
I'd _love_ to find a route around this 🙏